tilingManager/tilingLayouts/baseTiling.js

```diff
@@ -82,20 +84,76 @@ var BaseTilingLayout = class BaseTilingLayout {
 
     moveMetaWindow(metaWindow, x, y) {
         this.callSafely(metaWindow, metaWindowInside => {
-            metaWindowInside.move_frame(false, x, y);
+            metaWindowInside.move_frame(true, x, y);
         });
     }
 
     moveAndResizeMetaWindow(metaWindow, x, y, width, height) {
+        const gap = global.tilingManager.gap;
+        const tweenTime = global.tilingManager.tweenTime;
+        if (gap) {
+            x = x + gap / 2;
+            y = y + gap / 2;
+            width = width - gap;
+            height = height - gap;
+        }
+
+        const rect = metaWindow.get_frame_rect();
+        const buf = metaWindow.get_buffer_rect();
+        x = Math.floor(x);
+        y = Math.floor(y);
+        width = Math.floor(width);
+        height = Math.floor(height);
+        if (
+            x === rect.x &&
+            y === rect.y &&
+            width === rect.width &&
+            height === rect.height
+        ) {
+            return;
+        }
         this.callSafely(metaWindow, metaWindowInside => {
-            metaWindowInside.move_resize_frame(false, x, y, width, height);
+            const actor = metaWindowInside.get_compositor_private();
+            const oldRect = metaWindow.get_frame_rect();
+            const [px, py] = global.get_pointer();
+
+            if (metaWindow.grabbed) {
+                const aw = actor.width;
+                const ah = actor.height;
+                const grabX = (px - actor.x) / actor.width;
+                const grabY = (py - actor.y) / actor.height;
+                actor.set_pivot_point(grabX, grabY);
+                Tweener.addTween(actor, {
+                    scale_x: width / oldRect.width,
+                    scale_y: height / oldRect.height,
+                    time: tweenTime,
+                    transition: 'easeOutQuad'
+                });
+                return;
+            }
+
+            metaWindow.move_resize_frame(true, x, y, width, height);
+            const newRect = metaWindow.get_frame_rect();
+            actor.opacity = 255;
+            actor.scale_x = oldRect.width / newRect.width;
+            actor.scale_y = oldRect.height / newRect.height;
+            actor.translation_x = oldRect.x - newRect.x;
+            actor.translation_y = oldRect.y - newRect.y;
+            Tweener.addTween(actor, {
+                scale_x: 1.0,
+                scale_y: 1.0,
+                translation_x: 0,
+                translation_y: 0,
+                time: tweenTime,
```

Comment:
The animation speed should be relative to distance to travel to get constant speed. Otherwise short distance will feel very slow and far distance very fast.

Reply:
I'm not so sure about this one, gnome-shell use fixed times in all their tweening and I'm afraid it will introduce weirdness in some edge cases (like when several tweenings overwrite each other).

Reply:
Let's tweak it later if it's needed